src/internal/plugins/helpers/use-global-breadcrumbs.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import { awsuiPluginsInternal } from '../api';
+import { getGlobalFlag } from '../../utils/global-flags';
+import { BreadcrumbGroupProps } from '../../../breadcrumb-group/interfaces';
+import { BreadcrumbsGlobalRegistration } from '../controllers/breadcrumbs';
+
+function useSetGlobalBreadcrumbsImplementation(props: BreadcrumbGroupProps<any>) {
+  const registrationRef = useRef<BreadcrumbsGlobalRegistration<BreadcrumbGroupProps> | null>();
+  const [registered, setRegistered] = useState(false);
+
+  useEffect(() => {
+    const registration = awsuiPluginsInternal.breadcrumbs.registerBreadcrumbs(props, () => setRegistered(true));
+    registrationRef.current = registration;
+
+    return () => {
+      registration.cleanup();
+    };
+    // subsequent prop changes are handled by another effect
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useLayoutEffect(() => {
```

Comment:
Since `useLayoutEffect` can have a negative performance impact, (why) do we need to use it?

Reply:
It causes impact when touches the hosting component:


```
useLayoutEffect(() => {
   setState(something); // this is bad
}, [...])
```

Here we call an API outside React, no concerns here.